cpp/tests/io/json/json_test.cpp

```diff
@@ -2856,6 +2856,34 @@ TEST_F(JsonReaderTest, JSONMixedTypeChildren)
   }
 }
 
+TEST_F(JsonReaderTest, MixedTypesWithSchema)
```

Comment:
I feel this needs more tests given the number of untested code paths we could have

Reply:
The code is tested with all existing unit test cases (with experimental enabled as default for all cpp and python unit tests). But it's moved to experimental flag to not mess with other users since it is close to code freeze.
I will add a few more unit tests that's not covered by existing unit tests.